app/src/main/java/com/palone/planahead/screens/taskEdit/TaskEditViewModel.kt

```diff
@@ -0,0 +1,253 @@
+        val taskId = taskRepository.upsert(task)
+        addAlertsToDatabase(alertProperty, millisToTaskEvent, taskId)
+    }
+
+    private suspend fun addAlertsToDatabase(
+        alertProperty: List<AlertFieldProperty>,
+        millisToTaskEvent: Long,
+        taskId: Long,
+        interval: Long = 0,
+    ) {
+        alertProperty.forEach {
+            val alertMillisDuration =
+                Duration.of(
+                    it.valueToShowAlertBeforeTaskEvent.toLong(),
+                    it.unitToShowAlertBeforeTaskEvent
+                ).toMillis()
+            val millisToNotifyBeforeTaskEvent = millisToTaskEvent - alertMillisDuration
+            val alert = Alert(
+                taskId = taskId.toInt(),
+                alertTypeName = it.type,
+                alertTriggerName = AlertTrigger.TIME,
+                eventMillisInEpoch = millisToNotifyBeforeTaskEvent,
+                interval = interval
+            )
+            alertRepository.upsert(alert)
+        }
+    }
+
+    private suspend fun createCronDatabaseEntry(
+        typeProperties: CronProperties = _cronProperties.value,
+        alertProperty: List<AlertFieldProperty> = _alertProperties.value,
+        name: String
+    ) {
+        val todayDateTime = LocalDateTime.of(LocalDate.now(), typeProperties.dayTime)
+        val millisToTaskEvent = getEpochMillisFromLocalDate(todayDateTime)
+        val task = Task(
+            taskId = null,
+            taskType = TaskType.CRON,
+            description = name,
+            LocalDateTime.now().toString(),
+            isCompleted = false,
+            priority = TaskPriority.LOW
+        )
+        val taskId = taskRepository.upsert(task)
+        when (typeProperties.repeatMode) {
+            TaskRepeatMode.DAILY -> addAlertsToDatabase(
+                alertProperty,
+                millisToTaskEvent,
+                taskId,
+                Duration.of(1, ChronoUnit.DAYS).toMillis()
+            )
+
+            TaskRepeatMode.WEEKLY -> addWeeklyAlertsToDatabaseByDays(
+                taskId,
+                typeProperties.daysOfWeek,
+                typeProperties.dayTime,
+                alertProperty
+            )
+
+            TaskRepeatMode.MONTHLY -> TODO()
+        }
+    }
+
+    private suspend fun addWeeklyAlertsToDatabaseByDays(
+        taskId: Long,
+        daysOfWeek: List<DayOfWeek>,
+        dayTime: LocalTime,
+        alertProperty: List<AlertFieldProperty>
+    ) {
+        daysOfWeek.forEach {
+            val wantedDay = LocalDate.now().plusDays(getDaysToNextDayOfWeek(it).toLong())
+            val wantedDayTime = LocalDateTime.of(wantedDay, dayTime)
+            addAlertsToDatabase(
+                alertProperty,
+                getEpochMillisFromLocalDate(wantedDayTime),
+                taskId,
+                Duration.of(7, ChronoUnit.DAYS).toMillis()
+            ) // Weekly
+        }
+    }
+    
+    private fun getDaysToNextDayOfWeek(day: DayOfWeek): Int {
```

Comment:
This doesn't belong to the viewmodel - could be some helper service class. We can inject it into ViewModel through Dependency Injection: 

https://developer.android.com/training/dependency-injection/hilt-android